HttpServer.cs

```diff
@@ -140,6 +144,7 @@ class HttpServer {
 
   private void Process (HttpListenerContext context) {
     string filename = context.Request.Url.AbsolutePath;
+    filename = HttpUtility.UrlDecode(filename);
```

Comment:
we could make this one into

string filename = HttpUtility.UrlDecode(context.Request.Url.AbsolutePath);